design/automated-provisioning-design.md

```diff
@@ -0,0 +1,147 @@
+
+- If we implement managed Envoy, then the Gateway provisioner has less to do: provision the Contour, pass configuration along, and the Contour will provision Envoy.
+- If we don’t implement managed Envoy, then the Gateway provisioner implements the Envoy provisioning logic.
+- Either way, the logic needs to be implemented and live somewhere.
+
+Advantages of doing managed Envoy: 
+- Users who don’t want automated Gateway provisioning, but do want automated Envoy provisioning, can have it
+- Users who don’t want to use Gateway API can still take advantage of automated Envoy provisioning
+- Listener programming (combo of Envoy service + Envoy listeners) can be done in one place
+
+Disadvantages of doing managed Envoy:
+- Lots of code churn in Contour proper, potentially harder to isolate while still in development/alpha than if implemented in an entirely new component.
+
+## Alternatives Considered
+### Alternative 1
+We considered continuing to invest in the Contour Operator, including the Contour CRD, with an eye towards bringing it to beta and eventually GA, and implementing the Gateway provisioner within the Operator (since they would share much of the underlying logic).
+Our first challenge with this option is that we have failed to establish a community of contributors around the Operator, so the work would need to be done by the core Contour team of maintainers, which would detract from Contour development.
+Our second challenge is that we have seen and heard of only limited usage of the Operator in the wild, so it’s not clear to us that continuing to develop it is an important priority for our users.
+Finally, continuing to maintain the Operator in a separate repository creates development overhead, since various bits of code and configuration must be manually kept in sync between Contour and the Operator. 
+
+### Alternative 2
+We also considered converting the existing Contour Operator into a Gateway provisioner, dropping the Contour CRD, and only supporting a Gateway API-based dynamic provisioning workflow.
+The main drawback here is that the Gateway provisioner is in a separate repository from core Contour code, resulting in a need to sync code and configuration between them.
+We felt that this was an inferior choice to implementing the Gateway provisioner within the Contour repository, which simplifies the development and release workflows.
+
+### Alternative 3
+We considered adding a new binary to `projectcontour/contour` to implement the Gateway provisioner.
+However, this introduces a fair amount of overhead in churn in terms of new Makefile targets, a new Dockerfile, new steps in the CI machinery, etc.
+Given that the contour binary already has numerous subcommands (`serve`, `certgen`, etc.), it seemed like a reasonable extension of that pattern to add another subcommand for the Gateway provisioner, and it also saves a lot of work.
+
+## Implementation
+
+### Approach
```

Comment:
I really like this approach and this whole section, I think that making the PR reviewable is very important with a big change like this.